java/src/jmri/managers/DefaultShutDownManager.java

```diff
@@ -247,7 +247,7 @@ protected boolean shutdown(int status, boolean exit) {
             log.debug("windows completed closing {} milliseconds after starting shutdown", new Date().getTime() - start.getTime());
             // wait for parallel tasks to complete
             try {
-                if (!runnables.isEmpty() && !new ProxyTask(runnables.stream()
+                if (!runnables.isEmpty() && !new ProxyTask(new HashSet<>(runnables).stream()
                         .map(task -> RP.post(task, 0, Thread.currentThread().getPriority()))
                         .collect(Collectors.toSet()))
                                 .waitFinished(timeout)) {
```

Comment:
There is a lot going on on these lines and I find it difficult to read. A suggestion is to change it to:

```
Collection<? extends Task> c = new HashSet<>(runnables).stream()
        .map(task -> RP.post(task, 0, Thread.currentThread().getPriority()))
        .collect(Collectors.toSet());
if (!runnables.isEmpty() && !new ProxyTask(c).waitFinished(timeout)) {
```